src/main/java/uk/gov/hmcts/reform/sscs/config/CitizenCcdService.java

```diff
@@ -105,6 +137,64 @@ private CaseDetails newCase(SscsCaseData caseData, String eventType, String summ
         return caseDetails;
     }
 
+    @Retryable
+    public CaseDetails triggerCaseEventV2(String caseId, String eventType, String summary, String description, IdamTokens idamTokens) {
+        return updateCaseCitizenV2(caseId, eventType, idamTokens, data -> new UpdateResult(data, summary, description));
+    }
+
+    public record UpdateResult(SscsCaseData sscsCaseData, String summary, String description) { }
+
+    /**
+     * Update a case while making correct use of CCD's optimistic locking.
+     * Changes can be made to case data by the provided consumer which will always be provided
+     * the current version of case data from CCD's start event.
+     */
+    @Retryable
+    public CaseDetails updateCaseCitizenV2(String caseId, String eventType, IdamTokens idamTokens, Function<SscsCaseData, UpdateResult> newCaseData) {
+        log.info("Updating a draft with caseId {} and eventType {}, using updateCaseCitizenV2 method for citizen", caseId, eventType);
+        StartEventResponse startEventResponse = citizenCcdClient.startEventForCitizen(idamTokens, caseId, eventType);
+        var data = sscsCcdConvertService.getCaseData(startEventResponse.getCaseDetails().getData());
+        data.setWorkBasketFields(null); // This has to be done as the work basket fields default to a null instantiation and ccd views that as a create / update of which one of the fields the citizen doesn't have the authorisation to change
+
+        /**
+         * @see uk.gov.hmcts.reform.sscs.ccd.deserialisation.SscsCaseCallbackDeserializer#deserialize(String)
+         * setCcdCaseId & sortCollections are called above, so this functionality has been replicated here preserving existing logic
+         */
+        data.setCcdCaseId(caseId);
+        data.sortCollections();
+
+        var result = newCaseData.apply(data);
+        CaseDataContent caseDataContent = sscsCcdConvertService.getCaseDataContent(result.sscsCaseData, startEventResponse, result.summary, result.description);
+
+        return citizenCcdClient.submitEventForCitizen(idamTokens, caseId, caseDataContent);
+    }
+
+    @Retryable
+    public CaseDetails updateCaseCitizenV2(String caseId, String eventType, String summary, String description, IdamTokens idamTokens, Consumer<SscsCaseData> mutator) {
+        return updateCaseCitizenV2(caseId, eventType, idamTokens, data -> {
+            mutator.accept(data);
+            return new UpdateResult(data, summary, description);
+        });
+    }
+
+    @Retryable
+    public CaseDetails updateCaseCitizenV2(String caseId, String eventType, String summary, String description, IdamTokens idamTokens, UnaryOperator<SscsCaseData> newCaseData) {
```

Comment:
Can we reuse the existing updateCaseCitizenV2 using Consumer instead of UnaryOperator?

Reply:
Existing method cant be used as a new CaseData is returned with updated values instead of existing CaseData being updated.